.travis.yml

```diff
@@ -58,6 +58,19 @@ jobs:
         - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npx audit-ci --config ./audit-ci.json; fi
         - npm run lint
         - sudo env "PATH=$PATH" npm test
+    - stage: 'firmware linting and formatting checks'
+      language: python
+      before_install:
+        - wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+        - echo 'deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-12 main' | sudo tee -a /etc/apt/sources.list
+        - echo 'deb-src http://apt.llvm.org/xenial/ llvm-toolchain-xenial-12 main' | sudo tee -a /etc/apt/sources.list
+        - sudo apt-get update
+      install:
+        - sudo apt-get install -y clang-format-12
+        - sudo apt-get install cppcheck
+      script:
+        - cd firmware/boron-ins-fsm && python clang-format-check.py
+        - cppcheck src --enable=warning,style,performance,portability --error-exitcode=1
```

Comment:
Is this only doing the `/src` directory?

Reply:
It seems there are various issues associated with checking the test directory, claiming that there are too many #ifdef configurations and that it cannot extract a valid configuration, preventing cppcheck from running. Do you think running it on the test files is important? I personally think that the problems it will be looking for are more applicable to code rather than tests and that it isn't entirely necessary, especially with the issues - but let me know if you think otherwise

Reply:
I'm OK with this for now. Tests are code and can have similar problems as production code, but even without this for now it is a big improvement over what we had before. So no need to worry about it on your last day ;-)

Reply:
That's a good point, and thanks!